### ISSUE TITLE
Brouter Web version finds better route than Android version

### ISSUE THREAD

**Issue by Andreas2430** [abrensch/brouter#423]
Hi, 

while hiking I was wondering why Brouter in Locus suggests such a long detour from marker 1 to marker 2:
<img src="https://user-images.githubusercontent.com/102322614/163924131-38c17fc9-dadd-476c-b4eb-278eab994055.png" width="200px">

I thought that maybe the paths in OSM base data were not connected. However, when checking the route at home using the Brouter web version, it routes it just fine:
https://brouter.de/brouter-web/#map=18/47.84414/7.94329/osm-mapnik-german_style&lonlats=7.942085,47.844566;7.943223,47.845102

Tried different hiking and bike routing profiles on Android, but it never finds the short route that the web version finds. What could be wrong?

Kind regards
Andreas

**Comment by zod** [comment]
There can be several reasons (in decreasing probability)
- Routing segments out of date: The way was added about a year ago. If you haven't updated your data it doesn't contain the way
- Different profiles: You mention hiking, but the brouter-web link uses the trekking (bike) profile. LocusMap allows using either builtin profiles (made by @poutnikl) or default brouter profiles. Try using the same profile like in brouter-web after updating your data
- Different BRouter versions: There could be a different versions of BRouter used on android and brouter-web

**Comment by Andreas2430** [comment]
Hi zod, 

thanks for the ideas.

The map in Locus on Android is Openandromaps Germany, generated on April 01, 2022. So this shouldn't cause the issue.

The version of Brouter on Android is the latest version available in the Google Play Store (version 1.6.3, updated December 24, 2021, according to the description in the Google Play Store).

With all of the hiking and biking profiles on Brouter web, the route is generated correctly. 

With all of the hiking and biking profiles using Brouter in Locus on Android, the detour is added to the route as shown in the screenshot.

How do I distinguish in Locus whether a routing profile is an original Brouter profile or a profile by @poutnikl ?

**Comment by zod** [comment]
Routing data is different from map data. You can use the Download Manager within BRouter to update your routing data.

LocusMap classifies profiles as internal profiles or external profiles. I think by default only internal profiles are available.

**Comment by abrensch** [comment]
>> How do I distinguish in Locus whether a routing profile is an original
>> Brouter profile or a profile by [@poutnikl](https://github.com/poutnikl) ?

after using a profile in Locus, you should find it's source as "remote.brf" in BRouter's profile folder

**Comment by Andreas2430** [comment]
Thank you @zod, I had forgotten that Brouter routing data must be downloaded separately. I had mistakenly thought that Brouter would use the offline map data from Locus to route. 

Now I updated the routing data using Brouter's download manager, and with the new routing data the route is calculated correctly. Apparently I had downloaded the previous routing data a long time ago and thereafter never updated it, so it was different from the more current offline map data. 
Thanks!
